pkg/operator/ceph/cluster/crash/spec.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crash
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	opspec "github.com/rook/rook/pkg/operator/ceph/spec"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Use the same keyring as the mons
+	keyringStoreName = "rook-ceph-mons"
+)
+
+func (c *Crash) startCrashDaemonset(crashConfig *crashConfig) error {
+	ds := c.makeCrashDaemonSet(c.Namespace, crashConfig)
```

Comment:
Instead of a daemonset I'd propose that we start a deployment on each node where we need to collect the crash logs. Otherwise with the daemonset we would have pods running on all nodes even where ceph isn't running. We should be able to accomplish this by using the same node controller that is being used to start the disruption budget canary deployments on OSD nodes. We just need to select all ceph nodes instead of OSD nodes. @rohantmp What would it take to re-use this node controller?

Reply:
@rohantmp ? :)

Reply:
Really for the late reply... I'd started looking into this, but go sidetracked by another task and this fell through the cracks. Catching up and checking.

Reply:
We could definitely use the controller-runtime pattern, but the node-drain-canary controllers only creates one set of node-drain canaries per operator(in the operator namespace), and they have no association with a CephCluster. The clusterdisruption controller uses their status to glean status about the node.

We could certainly use a very similar pattern though. I'll hack together and push and example soon.